File: samples/file_center_of_mass/file_center_of_mass.py
#!/usr/bin/env python3

import json
import os

from kittycad import KittyCAD, KittyCADAPIError
from kittycad.models.file_import_format import FileImportFormat
from kittycad.models.unit_length import UnitLength

# Create a new client with your token parsed from the environment variable:
#   ZOO_API_TOKEN.
client = KittyCAD(timeout=500, verify_ssl=True)

# Read in the contents of the file.
file = open("./seesaw.obj", "rb")
# LITTERBOX-END-NON-EDITABLE-SECTION
content = file.read()
file.close()

try:
    fm = client.file.create_file_center_of_mass(
        output_unit=UnitLength.M,
        src_format=FileImportFormat.OBJ,
        body=content,
    )
except KittyCADAPIError as e:
    raise Exception(f"There was a problem: {e}")

if fm.center_of_mass is None:
    raise Exception("There was a problem")

print(f"File center of mass (meters): {fm.center_of_mass}")

partInfo = {
    "title": "output.json",
    "center_of_mass": fm.center_of_mass.model_dump(),
}

# LITTERBOX-START-NON-EDITABLE-SECTION
with open("output.json", "w", encoding="utf-8") as f:
    json.dump(partInfo, f, ensure_ascii=False, indent=4)
os.system("cp ./seesaw.obj ./output.obj")

File: samples/file_center_of_mass/file_center_of_mass.js
import fsp from 'fs/promises'
import { exec } from 'child_process'
import { promisify } from 'util'
import { file } from '@kittycad/lib/import'

async function main() {
    const body = await fsp.readFile('./seesaw.obj', 'base64')
    // LITTERBOX-END-NON-EDITABLE-SECTION

    const response = await file.create_file_center_of_mass({
        material_density: 7,
        src_format: 'obj',
        body,
    })
    if ('error_code' in response) throw response
    const { center_of_mass } = response
    console.log(`File center of mass: ${center_of_mass}`)
    const partInfo = {
        title: 'output.json',
        center_of_mass,
    }

    // LITTERBOX-START-NON-EDITABLE-SECTION
    await promisify(exec)('cp ./seesaw.obj ./output.obj')
    await fsp.writeFile('./output.json', JSON.stringify(partInfo, null, 2), 'utf8')
}

main()
